0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,47 +1,104 @@
+2. The receiver endpoint responds with the receiver info, including the recipient's ILP address.
+3. The sender choses an amount of their currency or assets to send.
+4. The sender's SPSP client uses its ILP module to quote how much of the recipient's currency or asset type will be delivered to the recipient's ILP address for the given source amount.
+5. The sender accepts the quote.
+6. The sender's SPSP client submits the payment information, including the destination amount, to the receiver endpoint.
+7. The receiver endpoint responds with an ILP Packet and Crypto Condition corresponding to the destination amount.
+8. The sender's SPSP client uses its ILP module to create a transfer **using the chosen source amount, NOT by quoting the ILP packet**, and attach the ILP packet to the transfer.
+9. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+10. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
+
+#### Invoice
+
+1. The sender's SPSP client queries the receiver endpoint.
+2. The receiver endpoint responds with the receiver info, including the invoice status, amount, and currency code.
+3. The sender's SPSP client submits the sender's info, including the sender's ILP address to the receiver endpoint.
+4. The receiver endpoint responds with an ILP Packet and Crypto Condition corresponding to the destination amount.
+5. The sender's SPSP client uses its ILP module to get a quote in their currency or asset type for the ILP transfer.
+6. The sender accepts the quote.
+7. The sender's SPSP client uses its ILP module to initiate the ILP transfer.
+8. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+9. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
+
+## Specification
+
+The receiver endpoint will respond to HTTP `GET` and `POST` requests in the following manner:
+
+### Query (`GET <receiver>`)
+
+The sender queries the receiver endpoint to get information about the type of payment that can be made to this receiver:
 
 ``` http
 GET /api/receivers/bob HTTP/1.1
```

Comment:
My initial answers:

1. Don't think so. 
2. Good question. I _think_ it's on the receiver to decide that.
3. Unless we have types aside from `payee` and `invoice`, no. If there are only these two I think it makes sense for an endpoint to be one or the other. If someone gives you an invoice receiver URL, then you'll use it as such. If you have a payee endpoint I can't think of a use case where you'd need to go from that to an invoice (because you'd get the invoice URL from some other context)

Another way to look at this though would be that the receiver can give you any valid URL that you can treat as opaque and it will work. I think I like that better.